pkg/adapter/adapter.go

```diff
@@ -31,17 +35,150 @@ type envConfig struct {
 	MongoDbCredentialsPath string `envconfig:"MONGODB_CREDENTIALS" required:"true"`
 	Database               string `envconfig:"MONGODB_DATABASE" required:"true"`
 	Collection             string `envconfig:"MONGODB_COLLECTION" required:"false"`
+	EventSource            string `envconfig:"EVENT_SOURCE" required:"false"`
+	SinkURL                string `envconfig:"K_SINK" required:"false"`
 }
 
-// NewEnvConfig creates an empty configuration
+type mongoDbAdapter struct {
+	namespace  string
+	ce         cloudevents.Client
+	client     *mongo.Client
+	source     string
+	datasource *dataSource
+	sinkurl    string
+	logger     *zap.SugaredLogger
+}
+
+// dataSource interface to interact with either a database or a collection.
+type dataSource interface {
+	Watch(ctx context.Context, pipeline interface{}, opts ...*options.ChangeStreamOptions) (*mongo.ChangeStream, error)
+}
+
+// NewEnvConfig creates an empty environement variables configuration.
 func NewEnvConfig() adapter.EnvConfigAccessor {
 	return &envConfig{}
 }
 
 // NewAdapter creates an adapter to convert incoming MongoDb changes events to CloudEvents and
-// then sends them to the specified Sink
+// then sends them to the specified Sink.
 func NewAdapter(ctx context.Context, processed adapter.EnvConfigAccessor, ceClient cloudevents.Client) adapter.Adapter {
-	fmt.Printf("Hey from mongodb adapter")
-	time.Sleep(600 * time.Second)
+	logger := logging.FromContext(ctx)
+	env := processed.(*envConfig)
+
+	rawURI, err := ioutil.ReadFile(env.MongoDbCredentialsPath + "/URI")
+	if err != nil {
+		logger.Desugar().Error("Unable to get MongoDb URI field", zap.Any("secretPath", env.MongoDbCredentialsPath+"/URI"))
+	}
+	URI := string(rawURI)
+	if URI == "" {
+		logger.Desugar().Error("MongoDb URI field is an empty string", zap.Any("URI", URI))
+	}
```

Comment:
IMO the receiver_adapter lib you're depending on should've put err as a return parameter. Logging error doesn't prevent your program from crashing if the sink URL is not set. See my other comment.

Reply:
True, what receive_adapter library you're talking about? I seems odd for it to be in receive_adapter.go

Reply:
move URI checks to Start to be able to return err